src/test/java/educationalproject/programmingstuff/tests_integrations/EndpointsProcessingTests.java

```diff
@@ -0,0 +1,171 @@
+        userEntities = userRepository.findAll();
+        userResponseDtos = userResponseMapper.makeUsersResponseOf(userEntities);
+        expectedResponse = objectMapper.writeValueAsString(userResponseDtos);
+
+        //When
+        ResultActions resultActions = mockMvc.perform(request);
+
+        //Then
+        resultActions
+                .andExpect(status().isOk())
+                .andExpect(content().json(expectedResponse));
+    }
+
+    @Test
+    @Transactional
+    void givenGetEndpointUsersWithName_whenEndpointTriggered_thenReturnUsersWithThisName() throws Exception {
+
+        //Given
+        String testName = "John";
+        request = MockMvcRequestBuilders.get("/users").param("name", testName);
+        userEntities = userRepository.getUsersByName(testName);
+        userResponseDtos = userResponseMapper.makeUsersResponseOf(userEntities);
+        expectedResponse = objectMapper.writeValueAsString(userResponseDtos);
+
+        //When
+        ResultActions resultActions = mockMvc.perform(request);
+
+        //Then
+        resultActions
+                .andExpect(status().isOk())
+                .andExpect(content().json(expectedResponse));
+    }
+
+    @Test
+    @Transactional
+    void givenPostEndpointUsersWithNewUser_whenEndpointTriggered_thenReturnThatUserWithNewId() throws Exception {
+
+        //Given
+
+        String testName = "Sasha";
+        String testSurname = "Grey";
+
+        RequestBuilder request = prepareRequest(testName, testSurname);
+
+        int usersTotalBeforeNewOneCreation = getAmountOfRegisteredUsers();
+
+        //When
+        MvcResult mvcResult = mockMvc.perform(request)
+                .andExpect(status().isOk())
+                .andReturn();
+
+        //Then
+        MockHttpServletResponse response = mvcResult.getResponse();
+        String content = response.getContentAsString();
+        UserResponseDto userResponseDto = objectMapper.readValue(content, UserResponseDto.class);
+
+        assertNotNull(userResponseDto.getId());
+        assertEquals(usersTotalBeforeNewOneCreation + 1, getAmountOfRegisteredUsers());
+        assertEquals(testName,userResponseDto.getUserName());
+        assertEquals(testSurname,userResponseDto.getSurname());
+        assertNull(userResponseDto.getOrders());
+
+    }
+
+    private RequestBuilder prepareRequest(String testName, String testSurname) throws JsonProcessingException {
+
+        UserCreateRequestDto userCreateRequestDto = UserCreateRequestDto.builder()
+                .userName(testName)
+                .surname(testSurname)
+                .build();
+
+        String body = objectMapper.writeValueAsString(userCreateRequestDto);
+
+        request = MockMvcRequestBuilders.post("/users")
+                .content(body)
+                .contentType(MediaType.APPLICATION_JSON);
+
+        return request;
+    }
+
+    private int getAmountOfRegisteredUsers() {
+
+        List<UserResponseDto> users = userService.getAllUsers();
```

Comment:
jpa repository should have a method like "contains" or smth, so you can just check whether there's a specified entity in the db or not